### PR TITLE
Adds in a fade in/out animation for chat + hover to show functionality

### LINKED ISSUE #8630: Make chat underlayer visible when hovering with mouse over it
There are many instances where chat is hard to read, mostly because the underlayer isn't visible during gameplay and the background is light. You then have to stop and press the chat key to read it.

This proposal of solution is to make it so chat detects the mouse entering its area and makes the underlay visible while the mouse is in.

### PR DESCRIPTION
**Pull request by MaxIsJoe**
closes #8630

# Changelog:

CL: [New] - The chat background is now animated when focusing/unfocusing.
CL: [Improvement] - Hovering your mouse over the chat will now show the background to allow players to more easily read text.
CL: [Improvement] - You can configure how transparent the chat background should be when unfocused.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Core/Chat/ChatUI.cs [hunk 1]
```diff
@@ -84,6 +84,11 @@ public class ChatUI : SingletonManager<ChatUI>
 		/// </summary>
 		public event System.Action OnChatWindowClosed;
 
+		[BoxGroup("Animation")] public float ChatFadeSpeed = 2f;
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/Chat/ChatUI.cs:87]: Is this a 2 seconds animation?

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/Chat/ChatUI.cs:87]: no, the speed is a weight of the lerp, not the speed of the animation

Comment (corp-0) [reply UnityProject/Assets/Scripts/Core/Chat/ChatUI.cs:87]: How does it look?

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/Chat/ChatUI.cs:87]: https://user-images.githubusercontent.com/34368774/174357213-1159f765-0a3b-460a-82e3-ad51ee6c133b.mp4

**Comment by PerfectTangent** [comment]
Great PR, much excite, thank you.

File: UnityProject/Assets/Scripts/Core/Chat/ChatUI.cs (hunk 1 above)
Comment (corp-0) [reply UnityProject/Assets/Scripts/Core/Chat/ChatUI.cs:87]: personally, I  think the fade-in is too slow for a default value, might just be me

**Comment by NoooneyDude** [comment]
Does this mean I will no longer be able to interact with the world behind the chat, when I don't have the chat open?

**Comment by MaxIsJoe** [comment]
> Does this mean I will no longer be able to interact with the world behind the chat, when I don't have the chat open?

shouldn't be, I'll test this just in case

File: UnityProject/Assets/Scripts/Core/Chat/ChatUI.cs (hunk 1 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Chat/ChatUI.cs:87]: Might to get annoying with the mouseover if you're holding a gun?

**Comment by Bod9001** [comment]
say when you've tested

**Review by Bod9001 — DISMISSED** [review]

**Comment by Fr-Dae** [comment]
while you have your hands in the code that concerns the chat, would it be possible to consider adding an option to facilitate readability?

as explained in the following conversation

thank you for your attention =)

https://github.com/unitystation/unitystation/discussions/8458

**Comment by MaxIsJoe** [comment]
> while you have your hands in the code that concerns the chat, would it be possible to consider adding an option to facilitate readability?
> 
> as explained in the following conversation
> 
> thank you for your attention =)
> 
> #8458

that requires more work than im willing to put currently

File: UnityProject/Assets/Scripts/Core/Chat/ChatUI.cs (hunk 1 above)
Comment (corp-0) [reply UnityProject/Assets/Scripts/Core/Chat/ChatUI.cs:87]: Can you make it last half the time it is taking in the video?

**Comment by MaxIsJoe** [comment]
Alright so im done testing and fixed the following :
Made the animation faster for Gillies
Made the raycast for the hover check be moved to the very edge of the screen rather than the entire chat window to avoid it getting in the way of combat
Fixed a weird issue with lerps making the entire alpha disappear when using high weights

**Comment by corp-0** [comment]
oh that detection change is clever, nice

**Review by corp-0 — DISMISSED** [review]

**Review by Bod9001 — DISMISSED** [review]

**Comment by MaxIsJoe** [comment]
hold on let me do one last check before i ask for approvals

**Comment by MaxIsJoe** [comment]
okay nevermind *now* it's fixed for real.
approvals pls.

**Review by corp-0 — APPROVED** [review]

**Comment by MaxIsJoe** [comment]
checks pass
meeeeeeeeeeeeeeeeeeeeerrrrgeeeeeeee

**Pull request merged**